From b83dc6241f7cd32412f194e0e1a0f3564a22c922 Mon Sep 17 00:00:00 2001
From: Stefan Chulski <stefanc@marvell.com>
Date: Wed, 18 Oct 2017 17:07:38 +0300
Subject: [PATCH 1242/1345] fix: net: mvpp2x: populate of_node of net_class
 device

commit  a2f3188c762f5911acc5f4a3e8977dbbad718c39 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

Patch fix DSA driver probe fail.
DSA driver failed to find netdev device by of_find_net_device_by_node
Callback, due missed of_node pointer in netdev parent device.

Fix:
Populate of_node of netdev parent device during port probe.

Change-Id: I557faf614a486b4268750920df3754b2eae50167
Signed-off-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/45329
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Yuval Caduri <cyuval@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
(cherry picked from commit bf5549e65cacc68804be262134d8dab099b2382c)
Reviewed-on: http://vgitil04.il.marvell.com:8080/45927
Reviewed-by: Yoav Gvili <ygvili@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
index 35f2bdb..2f28a91 100644
--- a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
@@ -5274,6 +5274,9 @@ static int mv_pp2x_port_probe(struct platform_device *pdev,
 	mv_pp2x_ptp_init(pdev, port, id);
 #endif
 
+	/* Populate network device of_node */
+	dev->dev.of_node = port_node;
+
 	return 0;
 	dev_err(&pdev->dev, "%s failed for port_id(%d)\n", __func__, id);
 
-- 
1.7.9.5

